servlet/src/main/java/com/linecorp/armeria/server/servlet/ServletRegistration.java

```diff
@@ -0,0 +1,252 @@
+            }
+
+            @Override
+            public Enumeration<String> getInitParameterNames() {
+                return Collections.enumeration(ImmutableSet.copyOf(getInitParameters().keySet()));
+            }
+        };
+    }
+
+    /**
+     * Get servlet security element.
+     */
+    @Nullable
+    public ServletSecurityElement getServletSecurityElement() {
+        return servletSecurityElement;
+    }
+
+    /**
+     * Get multipart config element.
+     */
+    @Nullable
+    public MultipartConfigElement getMultipartConfigElement() {
+        return multipartConfigElement;
+    }
+
+    /**
+     * Get servlet config.
+     */
+    public ServletConfig getServletConfig() {
+        return servletConfig;
+    }
+
+    /**
+     * Get servlet.
+     */
+    public Servlet getServlet() {
+        return servlet;
+    }
+
+    /**
+     * Is async supported.
+     */
+    public Boolean isAsyncSupported() {
+        return asyncSupported;
+    }
+
+    /**
+     * Get load on startup.
+     */
+    public int getLoadOnStartup() {
+        return loadOnStartup;
+    }
+
+    /**
+     * Is initialization servlet cas.
+     */
+    public boolean isInitServletCas(boolean expect, boolean update) {
+        return initServlet.compareAndSet(expect, update);
+    }
+
+    /**
+     * Is initialization servlet.
+     */
+    public boolean isInitServlet() {
+        return initServlet.get();
+    }
+
+    @Override
+    public Set<String> addMapping(String... urlPatterns) {
+        requireNonNull(urlPatterns, "urlPatterns");
+        mappingSet.addAll(Arrays.asList(urlPatterns));
+        for (String pattern : urlPatterns) {
+            urlMapper.addMapping(pattern, this, servletName);
+        }
+        return mappingSet;
+    }
+
+    @Override
+    public Collection<String> getMappings() {
+        return mappingSet;
+    }
+
+    @Override
+    @Nullable
+    public String getRunAsRole() {
+        return roleName;
+    }
+
+    @Override
+    public String getName() {
+        return servletName;
+    }
+
+    @Override
+    public String getClassName() {
+        return servlet.getClass().getName();
+    }
+
+    @Override
+    public boolean setInitParameter(String name, String value) {
+        throw new IllegalStateException("Can't set init parameter after ServletRegistration is initialized");
+    }
+
+    @Override
+    @Nullable
+    public String getInitParameter(String name) {
+        requireNonNull(name, "name");
+        return initParameterMap.get(name);
+    }
+
+    @Override
+    public Set<String> setInitParameters(Map<String, String> initParameters) {
+        throw new IllegalStateException("Can't set init parameters after ServletRegistration is initialized");
+    }
+
+    @Override
+    public Map<String, String> getInitParameters() {
+        return initParameterMap;
+    }
+
+    @Override
+    public void setLoadOnStartup(int loadOnStartup) {
+        this.loadOnStartup = loadOnStartup;
```

Comment:
Question: Don't we need to validate the value before assigning?

Reply:
I see method's document: loadOnStartup may be positive, negative and zero. Therefore, I don't validate this value.